src/renderer/components/dex/OrderForm.vue

```diff
@@ -368,23 +369,128 @@ export default {
     },
 
     percent(value) {
-      let unitPrice = this.$store.state.orderPrice;
-      if (this.orderType !== 'Limit') {
-        // todo, make smarter to calculate price based on depth of book
-        if (this.side === 'Buy') {
-          unitPrice = this.$store.state.orderBook.asks[0].price;
-        } else if (this.side === 'Sell') {
-          unitPrice = this.$store.state.orderBook.bids[0].price;
-        }
+      if (this.side === 'Buy') {
+        return this.percentForBuy(value);
       }
 
-      let newQuantity = 0;
-      if (this.side === 'Buy') {
-        newQuantity = (this.baseHolding.totalBalance / unitPrice) * value;
-      } else if (this.side === 'Sell') {
-        newQuantity = this.quoteHolding.totalBalance * value;
+      return this.percentForSell(value);
+    },
+
+    percentForBuy(value) {
+      if (!this.baseHolding || !this.baseHolding.availableBalance
```

Comment:
A good chunk of this method seems redundant with percentForSell.

I would like to see one or more common methods both percentForBuy and percentForSell call to reduce the copied code.

Reply:
I guess we can do it later.